core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -174,6 +192,26 @@ private[spark] class IndexShuffleBlockResolver(
     }
   }
 
+  private def getChecksums(checksumFile: File, blockNum: Int): Array[Long] = {
+    if (!checksumFile.exists()) return null
+    val checksums = new ArrayBuffer[Long]
+    // Read the checksums of blocks
+    var in: DataInputStream = null
+    try {
+      in = new DataInputStream(new NioBufferedFileInputStream(checksumFile))
+      while (checksums.size < blockNum) {
+        checksums += in.readLong()
+      }
+    } catch {
+      case _: IOException | _: EOFException =>
+        return null
+    } finally {
+      in.close()
+    }
+
```

Comment:
Something like this might be better ?

```suggestion
    Try(Utils.tryWithResource(new DataInputStream(new NioBufferedFileInputStream(checksumFile))) {
      in => Array.tabulate(blockNum)(_ => in.readLong())
    }).getOrElse(null)
```